src/metabase/db/custom_migrations.clj

```diff
@@ -105,6 +105,38 @@
   (when s
     (.toLowerCase (str s) Locale/US)))
 
+(defn json-in
+  "Default in function for columns given a Toucan type `:json`. Serializes object as JSON."
+  [obj]
+  (if (string? obj)
+    obj
+    (json/generate-string obj)))
+
+(defn- json-out [s keywordize-keys?]
+  (if (string? s)
+    (try
+      (json/parse-string s keywordize-keys?)
+      (catch Throwable e
+        (log/error e "Error parsing JSON")
+        s))
+    s))
+
+(def ^:private encrypted-json-in
+  (comp encryption/maybe-encrypt json-in))
+
+(defn- encrypted-json-out
+  [v]
+  (let [decrypted (encryption/maybe-decrypt v)]
```

Comment:
I'm surprised that this is outside the try, can you explain why that is? A comment in the code would be great.

Reply:
It doesn't look like you updated this in the final version

Reply:
I think it's fine to change this only here even if you don't want to update the app code (yet?)

Reply:
ah, just realized it's that way because decrypted is also used in the catch body.

Reply:
Ahh, I see it also does a try catch, and will just return the raw value if there was a failure. Seems like this could fail horribly if the encrypted string is all numeric characters then... 😬 

I see there is already a warning logged if the secret key is not preset, or valid. Perhaps we should just raise that log level?

Reply:
I don't like these non-standard "maybe" semantics one bit 😄